SabreTools.Serialization/Deserializers/AppPkgHeader.cs

```diff
@@ -0,0 +1,35 @@
+using System.IO;
+using SabreTools.IO.Extensions;
+using static SabreTools.Models.PlayStation4.Constants;
+
+namespace SabreTools.Serialization.Deserializers
+{
+    public class AppPkgHeader : BaseBinaryDeserializer<Models.PlayStation4.AppPkgHeader>
+    {
+        /// <inheritdoc/>
+        public override Models.PlayStation4.AppPkgHeader? Deserialize(Stream? data)
+        {
+            // If the data is invalid
+            if (data == null || !data.CanRead)
+                return null;
+
+            try
+            {
+                // Create a new app.pkg header to fill
+                var appPkgHeader = new Models.PlayStation4.AppPkgHeader();
+
+                // Try to parse the header
+                var header = data.ReadType<Models.PlayStation4.AppPkgHeader>();

```

Comment:
This won't do what you want. This will read, by default, using little-endian on the majority of platforms. Once rebased, you can look at SFO and SFB as examples of how to handle big-endian deserialization.